src/item-navigator.ts

```diff
@@ -0,0 +1,510 @@
+import {
+  css,
+  html,
+  LitElement,
+  customElement,
+  property,
+  state,
+  query,
+  PropertyValues,
+  CSSResult,
+} from 'lit-element';
+import { nothing, TemplateResult } from 'lit-html';
+import { MetadataResponse } from '@internetarchive/search-service';
+import {
+  SharedResizeObserver,
+  SharedResizeObserverResizeHandlerInterface,
+} from '@internetarchive/shared-resize-observer';
+import '@internetarchive/ia-menu-slider';
+
+import { ModalManagerInterface } from '@internetarchive/modal-manager';
+import '@internetarchive/icon-ellipses/icon-ellipses';
+import './loader';
+
+import {
+  IntManageSideMenuEvent,
+  IntSetOpenMenuEvent,
+  IntSetMenuContentsEvent,
+  IntSetMenuShortcutsEvent,
+  IntLoadingStateUpdatedEvent,
+  IntManageFullscreenEvent,
+} from './interfaces/event-interfaces';
+
+import {
+  IntMenuProvider,
+  IntMenuShortcut,
+  MenuId,
+} from './interfaces/menu-interfaces';
+import './no-theater-available';
+
+export enum ItemType {
+  BOOK = 'bookreader',
+}
+@customElement('ia-item-navigator')
+export class ItemNavigator
+  extends LitElement
+  implements SharedResizeObserverResizeHandlerInterface {
+  @property({
+    type: Object,
+    converter: (value: string | MetadataResponse | null): MetadataResponse => {
+      if (value && typeof value === 'string') {
+        return new MetadataResponse(JSON.parse(atob(value)));
+      }
+      return value as MetadataResponse;
+    },
+  })
+  item?: MetadataResponse;
+
+  @property({ type: String }) itemType?: ItemType;
+
+  @property({ type: String }) baseHost = 'archive.org';
+
+  @property({
+    converter: (arg: string | boolean | null) => {
+      if (typeof arg === 'boolean') {
+        return arg;
+      }
+      return arg === 'true';
+    },
+  })
+  signedIn = false;
+
+  @property({ type: Array }) menuContents: IntMenuProvider[] = [];
+
+  @property({ type: Array }) menuShortcuts: IntMenuShortcut[] = [];
+
+  @property({ type: Boolean, reflect: true, attribute: true })
+  viewportInFullscreen: boolean | null = null;
+
+  @property({ type: Boolean }) menuOpened = false;
+
+  @property({ type: String }) openMenu: MenuId | '' = '';
```

Comment:
thanks for the push on this. i've got it as 

```
openMenu?: MenuId;
```
with event handler update to return `undefined` if menu is toggled
+ new `get this.selectedMenuId` where it will serve empty string as required by side menu component to close side panel.